Add login page and related assets

Alleyne-Angel opened the pull request:
close https://github.com/osuarena/osuarea-frontend/issues/22

Add pages folder, including login pages, following AelSan's design, and assets folder for the resources. 

And revert wrongly .gitignore edits

Alleyne-Angel committed:
Add login page and related assets

Introduced a new login page with HTML, CSS, and an empty JS file. Added various UI assets including icons and images required for the login interface.

cdwcgt commented:
要不要把入口调至login.html
还是说暂时现在main.html添加自动跳转到login的逻辑
然后以后再改?

RushFTK commented:
建议`login.css`->`style.css`
如果你用了公用的js文件，例如`bootstrap.js`，放在`../deps`里，否则放`deps`里即可
(自己写的这段`login.js`留在这里即可)

Alleyne-Angel committed:
Add lobby page with layout, styles, and navigation

Introduces a new lobby page with HTML structure, dedicated CSS for layout and navigation, and a JS file with placeholder functions for advertisements and tournaments. Updates main.js to load the new lobby page on app start. Also adds a missing global stylesheet for consistent styling across the app.

cdwcgt commented:
那我login.js这块？

cdwcgt commented:
{backend_endpoint}/auth/get_auth_link 会给予一个指向 osu.ppy.sh 的链接
需要用户通过外部浏览器打开登陆
然后需要绑定一个回调 将code和state回传到 {backend_endpoint}/auth/osu_callback 然后会给予 access_token ，根据是否记住登录状态来保存

see https://github.com/osuarena/osuarea-backend/blob/921959ba9416c4dbe50133c05b029473dc82b260/src/auth/auth.controller.ts#L26
